lib/stagehand/staging/checklist.rb

```diff
@@ -12,14 +12,16 @@ def self.related_commit_ids(commit)
         related_entries(commit.entries).collect(&:commit_id).select(&:present?).uniq
       end
 
-      def self.related_entries(entries)
+      def self.related_entries(entries, relation_filter = nil)
         entries = Array.wrap(entries)
         related_entries = []
 
         entries_to_spider = Array.wrap(entries)
         while entries_to_spider.present?
-          contained_matching = CommitEntry.contained.matching(entries_to_spider)
-          matching_commit_entries = CommitEntry.where(:commit_id => contained_matching.select(:commit_id))
+          contained_matching = CommitEntry.contained.matching(entries_to_spider).to_a
+          contained_matching.select! {|entry| relation_filter.call(entry) } if relation_filter.present?
```

Comment:
Since we're passing the entry directly to the relation filter, can we do `.select!(&relation_filter)`?